addons/survey/tests/test_survey_results.py

```diff
@@ -0,0 +1,279 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.survey.controllers.main import Survey
+from odoo.addons.survey.tests import common
+from odoo.osv import expression
+from odoo import fields
+
+
+class TestSurveyResults(common.TestSurveyCommon):
+
+    def setUp(self):
+        self.certification_survey_id = self.ref('survey.vendor_certification')
+        self.quiz_survey_id = self.ref('survey.survey_demo_quiz')
+        self.UserInput = self.env['survey.user_input']
+        self.UserInputLine = self.env['survey.user_input.line']
+        self.QuestionAnswer = self.env['survey.question.answer']
+
+    def _check_filters_results(self, post, survey_id, lines_domain):
+
+        with self.assertQueryCount(1):
+            base_domain = Survey._get_results_page_domain(survey_id, **post)
+            expected_user_input_line_ids = Survey._get_matching_user_input_lines(self, lines_domain, base_domain)
+
+        with self.assertQueryCount(1):
+            found_user_input_line_ids, _ = Survey._extract_filters_data(self, survey_id, post)
+
+        self.assertEqual(expected_user_input_line_ids, found_user_input_line_ids)
+
+    def test_get_matching_user_input_lines(self):
+        """
+            Check that we retrieve the accurate user input lines depending on the domain
+            Filters:
+                A,68,70 (matrix) : How often should you water those plants | Cactus : Once a month
+                A,0,57  (simple_choice) : Which of the following would you use to pollinate : Bees
+                L,0,107 (char_box) : What is your email ? : mark.brown23@example.com
```

Comment:
Hard coding ids is not a good practice. (They can change when you restart your server for repeating the same test without rebuilding the whole database for example)
It is better to first fetch them, then use them to build your filter strings, domains, expected result etc.
Ex, 
```py
how_often_should_you_water_id = self.ref('survey.survey_demo_quiz_p3_q5').id
cactus_row_id = self.ref('survey.survey_demo_quiz_p3_q5_row1').id
once_a_month_id = self.ref('survey.survey_demo_quiz_p3_q5_sug1').id
```

You can also define the ones you use different tests in `setUp` of course.